Open to external (#14)
SparshithNR/doc-tester: README.md

# DocTester

This library is used to run code samples in markdown documents as acceptance tests, using simple comment-based assertions. By testing code snippets in documentation, you can have greater confidence that code in your documentation works as advertised.

## Installation
```sh

> npm install sparshithNR/doc-tester#master --save-dev

Reviewer: you should likely just do `yarn add doc-tester` or `npm install doc-tester` and release the project to NPM.

# DocTester

> This library is used to test the code samples in the docments. Using this library you can always ensure that code samples in docments always work and avoids issues created for document update.

Reviewer 2: ```suggestion
This library is used to run code samples in markdown documents as acceptance tests, using simple comment-based assertions. By testing code snippets in documentation, you can have greater confidence that the code works as advertised.
```

## Installation
```sh
npm install sparshithNR/doc-tester#master --save-dev
```
## Usage
### From commandline
```sh
node_modules/.bin/doc-tester
```
#options
1. -f (--file)

> > File name/path which is to be tested. Defaulted to `README.md` in the root.

Reviewer 2: ```suggestion
> Path of file to be tested (default: `./README.md`).
```

2. -d (--debug)

> > Setting this to true will not remove the test file generated by parsing the documentation file. Defaulted to `false`.

Reviewer 2: I'd rename this option, since `node --debug` has a completely different meaning (legacy version of `node --inspect`), which may cause confusion.

Maybe `--cleanup`?

3. --inspect (--inspect-brk)

> > This will enable you to put debugging break points in test file. Defaulted to `false`.

Reviewer 2: - [ ] add link to [node inspector](https://nodejs.org/en/docs/guides/debugging-getting-started/#enable-inspector
```suggestion
> Runs tests w/ the node `--inspect` option, allowing a debugger to be attached. (default: `false`)
```

4. -o (--output)

> > Path/name where you want test file generated need to be placed once code is generated. Default path/name `test.js` and created in root directory.

Reviewer 2: ```suggestion
> Path where generated test file will be written. (default: `./test.js`)
```

### From code
```js
import { runTest } from 'doc-tester';
await runTest({
  codeArray: ['add(3,4) // equals: 7;'],
  importsArray: [`import { add } from './add'`]
} /* , options */);
```
### Options
1. testName
> Name for the test block. Defaulted to `Doc Test`.
2. debug
> Setting this to true will not remove the test file generated by parsing the documentation file. Defaulted to `false`.
3. inspect
> This will enable you to put debugging break points in test file. Defaulted to `false`.

> 4. testFile

Reviewer 2: I would make these consistent w/ the CLI flags above (i.e., `testFile` here and `--output` via the CLI is forcing your users to remember two different sets of option names).

To avoid duplicate stuff in your README, you may just want to direct readers to one source of truth

i.e., 

```
## CLI Options

### `--output`
Path where generated test file will be written. (default: `./test.js`)

## From Code

### Options

* [`output`](#--output)

```

You'll need to figure out how github defines anchors to get this stuff right,
